### ISSUE TITLE
Rudimentary S1-S16 keyboard for recording patterns

### ISSUE THREAD

**Issue by dComposer** [westlicht/performer#126]
I was thinking the other night while programming a pattern with the encoder just how much quicker it would be if S1-S16 acted like a piano keyboard when I was in record mode (and I didn't have a midi controller handy). S9 would be C, S1 would be C#, S10 would be D, S2 would be D#, S11 would be E, etc. maybe the lights could be lit in the pattern of the keyboard to give some sort of visual:

![Screen Shot 2019-08-12 at 10 21 05 PM](https://user-images.githubusercontent.com/53159438/62910709-94525300-bd4f-11e9-8f50-30e59d4bd027.png)

**Comment by westlicht** [comment]
I don't like the step keys to be reassigned for that by default. Having one octave only is also a bit limiting. Thats why you can attach a small midi controller. Not sure how to go about this.

**Comment by toneburst** [comment]
I guess you'd have to select the steps first, then drop into another mode to enter notes with the "keyboard".

The two "blank" buttons S3, S7 could be used to inc/dec the octave.

**Comment by toneburst** [comment]
From a hardware perspective, different switch cap colours/shades could be used to mark out the "black" and "white" keys.

I think the screen would have to show something to indicate the buttons were operating in keyboard, rather than step-selection mode.

**Comment by westlicht** [comment]
I'm just not too convinced of this idea yet. One option would be to always switch into this keyboard mode when recording is activated. But then there would be the need for yet another option to enable this behaviour, as by default it interferes with existing functionality. It just adds another layer of complexity that I'm not yet sure of its value. The existing methods for entering notes are already plentyful in my opinion, and superior if you don't mind attaching an external keyboard.

**Comment by dComposer** [comment]
Yeah this isn't meant to be better than using an external keyboard or anything, just something to sketch an idea with when you don't have an external keyboard handy. Many melodies are within an octave anyway, but I could foresee repurposing the previous/next buttons for a really simple octave shift since I never use these buttons and use the encoder to navigate between parameters usually.

Also, the step buttons aren't used at all for midi tracks, right? This could do double duty there as well for auditioning arpeggiator patterns or if you just want the ability to play a simple melody for a gig but not carry around a keyboard.

**Comment by ghztomash** [comment]
Also recording gates by tapping out rhythms on the T1-T8 keys would be pretty nice
